### PR TITLE
add metadata auth framework.

### PR DESCRIPTION
**Pull request by tcnichol**
Modeled after Max's pull request. Checks authorization of metadata based on either file or dataset.

### CODE REVIEW

**Comment by ddey2** [comment]
Need to be updated after #354 gets merged

**Comment by tcnichol** [comment]
Will update once that is merged.

File: backend/app/routers/authorization.py [hunk 1]
```diff
@@ -103,3 +104,13 @@ async def get_file_role(
 ):
     """Retrieve role of user for an individual file. Role cannot change between file versions."""
     return role
+
+
+@router.get("/metadata/{metadata_id}/role", response_model=AuthorizationMetadata)
+async def get_metadata_role(
+    metadata_id: str,
+    current_user=Depends(get_current_username),
+    role: RoleType = Depends(get_role_by_file),
```
Comment (max-zilla) [inline backend/app/routers/authorization.py:113]: should this be `get_role_by_metadata`?

File: backend/app/routers/metadata_files.py [hunk 2]
```diff
@@ -382,6 +382,8 @@ async def get_file_metadata(
         metadata = []
         async for md in db["metadata"].find(query):
             md_out = MetadataOut.from_mongo(md)
+            resource_type = md_out.resource.collection
+            resource_id = md_out.resource.resource_id
```
Comment (max-zilla) [inline backend/app/routers/metadata_files.py:386]: are these used?

**Review by max-zilla — COMMENTED** [review]
a couple comments for clarification

File: backend/app/routers/authorization.py (hunk 1 above)
Comment (tcnichol) [reply backend/app/routers/authorization.py:113]: fixed.

File: backend/app/routers/metadata_files.py (hunk 2 above)
Comment (tcnichol) [reply backend/app/routers/metadata_files.py:386]: removed

**Review by max-zilla — APPROVED** [review]

**Pull request merged**